src/plan_ordered_append.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+
+#include <postgres.h>
+#include <nodes/relation.h>
+#include <nodes/value.h>
+#include <optimizer/paths.h>
+#include <optimizer/pathnode.h>
+#include <optimizer/tlist.h>
+#include <utils/builtins.h>
+#include <utils/typcache.h>
+
+#include "compat.h"
+#include "guc.h"
+#include "hypertable.h"
+#include "hypertable_restrict_info.h"
+#include "constraint_aware_append.h"
+#include "planner.h"
+#include "plan_ordered_append.h"
+
+/*
+ * Check if conditions for doing ordered append optimization are fulfilled
+ */
+bool
+ts_ordered_append_do_optimization(PlannerInfo *root, RelOptInfo *rel, Hypertable *ht, bool *reverse)
+{
+	/* check if optimizations are enabled */
+	if (ts_guc_disable_optimizations || !ts_guc_ordered_append)
+		return false;
+
+	/*
+	 * only do this optimization for hypertables with 1 dimension and queries
+	 * with an ORDER BY and LIMIT clause
+	 */
+	if (ht->space->num_dimensions != 1 || root->parse->sortClause == NIL || root->limit_tuples == -1.0)
```

Comment:
Shouldn't you reject GROUP By cases as well?

Reply:
Why?

Reply:
Haven't seen any tests with GROUP BY